File: src/main/java/pl/patrykdolata/lifemanager/domain/AccountEntity.java
package pl.patrykdolata.lifemanager.domain;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.math.BigDecimal;

@Data
@Entity
@Table(name = "account")
public class AccountEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Size(min = 3, max = 120)
    @Column(name = "name", nullable = false, length = 120)
    private String name;

    @Column(name = "balance")
    private BigDecimal balance;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private UserEntity user;
}

File: src/main/kotlin/pl/patrykdolata/lifemanager/domain/AccountEntity.kt
package pl.patrykdolata.lifemanager.domain

import pl.patrykdolata.lifemanager.domain.AccountEntity.Companion.TABLE_NAME
import java.math.BigDecimal
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table
import javax.validation.constraints.Size

@Entity
@Table(name = TABLE_NAME)
class AccountEntity(user: UserEntity) : AbstractUserDependEntity<Long>(user) {

    companion object {
        const val TABLE_NAME = "account"
    }

    @Size(min = 3, max = 120)
    @Column(name = "name", length = 120)
    var name: String = ""

    @Column(name = "balance")
    var balance: BigDecimal = BigDecimal.ZERO
}
